services/121-service/src/registration/registrations.controller.ts

```diff
@@ -707,4 +708,42 @@ export class RegistrationsController {
       params.paId,
     );
   }
+
+  // This "wildcard" endpoint needs to be at the bottom of the file to avoid conflicts with other endpoints
+  @ApiTags('programs/registrations')
+  @AuthenticatedUser({ permissions: [PermissionEnum.RegistrationREAD] })
+  @ApiOperation({
+    summary:
+      '[SCOPED] Get a specific registration view based on registration id.',
+  })
+  @ApiParam({
+    name: 'programId',
+    required: true,
+    type: 'integer',
+  })
+  @ApiParam({
+    name: 'id',
+    required: true,
+    type: 'integer',
+  })
+  @Get('programs/:programId/registrations/:id')
+  public async findOne(
+    @Param('programId', ParseIntPipe) programId: number,
+    @Param('id', ParseIntPipe) id: number,
+  ): Promise<MappedPaginatedRegistrationDto> {
+    const registrationEntity =
+      await this.registrationsService.getPaginateRegistrationById({
+        id,
```

Comment:
@PeterSmallenbroek FYI I took an executive decision and went ahead and already made the change to use registration id, with the assumption that this is the direction we're headed. We can always switch to paID / programRegistrationId if we have to.